mock/mock.go

```diff
@@ -218,16 +218,22 @@ func (c *Call) Unset() *Call {
 
 	foundMatchingCall := false
 
-	for i, call := range c.Parent.ExpectedCalls {
+	// in-place filter slice for calls to be removed - iterate from 0'th to last skipping unnecessary ones
+	var index int // write index
+	for _, call := range c.Parent.ExpectedCalls {
```

Comment:
modify looks good to me

Reply:
thank you

Reply:
Hello, @felixliou!
Perhaps I misunderstood you. 
Do you mean "looks ok now, and ready for merge" or you ask me modify the code?
If so, could you pls clarify which modifications are required? Thank you.

Reply:
looks ok now, and ready for merge，but I have not authorized to merge this PR.

Reply:
Thank you for your feedback. Do you know anybody (possibly maintainers?) who can do it? Could you ping they please?

Reply:
Hi，
@ernesto-jimenez 
@matryer 
@boyan-soubachov 

Please check this pull request.

Thank you very much